CI: iOS UI Review screenshot suite times out on main after #238

## What happened
The post-merge `iOS UI Review (main)` workflow failed on main for the merge of PR #238.

- Workflow run: https://github.com/ganesh47/mather/actions/runs/24623489050
- Failing job: `UI Screenshot Review`
- Failure mode: workflow step timeout, not a product assertion

## Evidence
The failing step log ends with:

```
##[error]The action 'Run UI screenshot tests' has timed out after 15 minutes.
```

Earlier screenshot tests in the same job were still passing, including:
- `testScreenshot_ParentSummary`
- `testScreenshot_PilotRunbook`
- `testScreenshot_SessionConfig`
- `testScreenshot_Settings_EnableVS1`

The timeout hit as `testSessionHistoryShowsMultipleSavedSessionsInParentSummaryAndSettings` started, which suggests total suite duration drift rather than a crisp app crash.

## Why this matters
Main is green on CI/CodeQL/Scorecards, but the UI review lane is now red and should either be sped up, split, or given an explicit larger timeout so visual review stays trustworthy.

## Possible fixes
- Trim or split the screenshot suite so the workflow stays under the current step timeout.
- Make the long-running screenshot/history test non-blocking or move it to a separate workflow.
- Raise the workflow step timeout if the current suite duration is expected.

--- a/.github/workflows/ios-ui-review-main.yml
+++ b/.github/workflows/ios-ui-review-main.yml
@@ -95,11 +95,15 @@ jobs:
             ci-videos/ui-tests.mp4 &
           VIDEO_PID=$!
 
+          # Keep screenshot review focused on artifact-producing flows.
+          # Slower behavioral UI assertions belong in the main test lane.
           xcodebuild test-without-building \
             -project Mather.xcodeproj \
             -scheme Mather \
             -destination "$SIM_DESTINATION" \
-            -only-testing MatherUITests \
+            -only-testing MatherUITests/ScreenshotTests \
+            -skip-testing MatherUITests/ScreenshotTests/testConcreteBuild_AccentRowLockedUntilWarmFull \
+            -skip-testing MatherUITests/ScreenshotTests/testSessionHistoryShowsMultipleSavedSessionsInParentSummaryAndSettings \
             -resultBundlePath UITestResults.xcresult \
             CODE_SIGNING_ALLOWED=NO
           XCODE_EXIT=$?

